scipy/stats/_multivariate.py

```diff
@@ -3850,3 +3852,448 @@ def rvs(self, dim, size=1, random_state=None):
+        M, n = M[..., np.newaxis], n[..., np.newaxis]
+        mu = n*(m/M)
+        return self._checkresult(mu, mncond, np.NAN)
+
+    def var(self, m, n):
+        """
+        Variance of the Multivariate Hypergeometric distribution.
+
+        Parameters
+        ----------
+        %(_doc_default_callparams)s
+
+        Returns
+        -------
+        cov : array_like
+            The covariance matrix of the distribution
+        """
+        M, m, n, mncond = self._process_parameters(m, n)
+        M, n = M[..., np.newaxis], n[..., np.newaxis]
+        output = n * m/M * (M-m)/M * (M-n)/(M-1)
+        return self._checkresult(output, mncond, np.NAN)
+
+    def cov(self, m, n):
+        """
+        Covariance matrix of the Multivariate Hypergeometric distribution.
+
+        Parameters
+        ----------
+        %(_doc_default_callparams)s
+
+        Returns
+        -------
+        cov : array_like
+            The covariance matrix of the distribution
+        """
+        # cov( x_i,x_j ) = -n * (M-n)/(M-1) * (K_i*K_j) / (M**2)
+        M, m, n, mncond = self._process_parameters(m, n)
+        M = M[..., np.newaxis, np.newaxis]
+        n = n[..., np.newaxis, np.newaxis]
+        output = (-n * (M-n)/(M-1) / (M**2) *
+                  np.einsum("...i,...j->...ij", m, m))
+        dim = m.shape[-1]
+        M, n = M[..., 0, 0], n[..., 0, 0]
+        # diagonal entries need to be computed differently
+        for i in range(dim):
+            output[..., i, i] = (n * (M-n)/(M-1) *
+                                 m[..., i]*(M-m[..., i]) / (M**2))
+
+        return self._checkresult(output, mncond, np.NAN)
+
+    def rvs(self, m, n, size=1, random_state=None):
+        """
+        Draw random samples from a Multivariate Hypergeometric distribution.
+
+        Parameters
+        ----------
+        %(_doc_default_callparams)s
+        size : integer or iterable of integers, optional
+            Number of samples to draw (default 1).
+        %(_doc_random_state)s
+
+        Returns
+        -------
+        rvs : array_like
+            Random variates of shape (`size`, `len(p)`)
+
+        Notes
+        -----
+        %(_doc_callparams_note)s
+        """
+        M, m, n, mncond = self._process_parameters(m, n)
+
+        random_state = self._get_random_state(random_state)
+        try:
+            # Generator is only available in numpy >= 1.17 while
+            # `multivariate_hypergeometric` distribution is only
+            # available in numpy >= 1.18
+            if (isinstance(random_state, np.random.Generator) and
+                np.__version__ >= '1.18'):  # noqa: E129
+                out = random_state.multivariate_hypergeometric(m, n,
+                                                               size)
+                return _squeeze_output(out)
+        except AttributeError:
+            pass
+
+        if isinstance(size, int):
+            size = (size, )
```

Comment:
I don't have NumPy >= 1.18 right now, but presumably `Generator` does not care if `size` is a scalar? (Otherwise this would need to go above the `try` block, right?)

Reply:
The `size` is converted to `tuple` to initialize the `rvs` array correctly. If I am not wrong, the `size` parameter is expected to be an integer or a tuple. If that isn't the case, more conditions and error checking is required.

Reply:
> The `size` is converted to tuple to initialize the `rvs` array correctly

I understand. But you do this _after_ the code for performing the sampling with a `Generator`. If `Generator` allows `size` to be a scalar, this is fine. If `Generator`, like `RandomState`, requires `size` to be a tuple, you would want to move this code up.

Does that make sense?